src/Controller/UserController.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Comment;
+use App\Entity\Post;
+use App\Entity\User;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\File\Exception\AccessDeniedException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Class UserController
+ * @package App\Controller
+ * @Route("/api/users")
+ */
+class UserController extends AbstractController
+{
+    private EntityManagerInterface $entityManager;
+    private UserRepository $userRepository;
+    private SerializerInterface $serializer;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        UserRepository $userRepository,
+        SerializerInterface $serializer
+    )
+    {
+        $this->entityManager = $entityManager ;
+        $this->userRepository = $userRepository ;
+        $this->serializer = $serializer ;
+    }
+
+    /**
+     * @return JsonResponse
+     * @Route(name="api_users_collection_get", methods={"GET"})
+     */
+    public function collection(): JsonResponse
+    {
+        $users = $this->userRepository->findAll();
+        return $this->json($users);
+    }
+
+    /**
+     * @Route("/{id}", name="api_users_item_get", methods={"GET"})
+     * @param User $user
+     * @return JsonResponse
+     */
+    public function item(User $user): JsonResponse
+    {
+        return $this->json($user);
+    }
+
+    /**
+     * @Route(name="api_users_collection_post", methods={"POST"})
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function post(Request $request): JsonResponse
+    {
+        $securityContext = $this->container->get('security.authorization_checker');
+        if (!$securityContext->isGranted('IS_AUTHENTICATED_FULLY')) {
+            $user = $this->serializer->deserialize($request->getContent(), User::class, "json");
+            $this->entityManager->persist($user);
+            $this->entityManager->flush();
+
+            return $this->json($user, 201);
+        } else {
```

Comment:
De manière général, il faut éviter les `else`.

Dans ton cas tu aurais pu faire ça : 

```suggestion
        if ($securityContext->isGranted('IS_AUTHENTICATED_FULLY')) {
            throw $this->createAccessDeniedException('Vous êtes déjà connecté !');
        }
        
        $user = $this->serializer->deserialize($request->getContent(), User::class, "json");
        $this->entityManager->persist($user);
        $this->entityManager->flush();
        
        return $this->json
```

Reply:
Oui, en gros tu as le concept d``early return`.
Si ton if throw une exception ou return quelque chose, pas besoin de faire un else derrière, puisque ce code ne sera jamais appelé si tu passes dans la condition :)